### PR TITLE
Update protocol to support pluggable devices

### PR DESCRIPTION
**Pull request by fstasi**
Breaking: This PR addresses compatibility issues with the new pluggable monitor implemented in Arduino IDE2.

In order to support pluggable devices, the Protocol must change accordingly:
`pluggableMonitorSettings` is used for actual monitor/CLI 
`monitorUISettings` is used for UI only information

### CODE REVIEW

**Review by davegarthsimpson — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -16,57 +16,149 @@ The application is designed to be as agnostic as possible regarding how and wher
...
+    {
+      command: "CHANGE_SETTINGS",
+      data: Partial<MonitorSettings> // see section below
+    }
+    ```
 
-**NOTE: For performance sake, the raw data coming from the serial port that is sent by the middleware to the serial plotter has to be a stringified array of values, rather than the Command/Data object**
+#### Monitor Settings
+
+Settings changes, sent and received in the Serial Plotter App, must follow the following object structure
```
Comment (AlbyIanna) [inline README.md:85]:
```suggestion
Settings changes, sent and received in the Serial Plotter App, must have the following object structure
```

File: README.md [hunk 2]
```diff
@@ -16,57 +16,149 @@ The application is designed to be as agnostic as possible regarding how and wher
...
+
+```
+Partial<MonitorSettings> = {
+  pluggableMonitorSettings: PluggableMonitorSettings;
+  monitorUISettings: Partial<MonitorModelState>;
+}
+```
 
+That means a Setting Message can container `pluggableMonitorSettings` and/or `monitorUISettings`.
```
Comment (AlbyIanna) [inline README.md:94]:
```suggestion
That means a Setting Message can contain `pluggableMonitorSettings` and/or `monitorUISettings`.
```

**Review by AlbyIanna — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -1,8 +1,7 @@
 {
   "name": "arduino-serial-plotter-webapp",
-  "version": "0.0.17",
-  "dependencies": {
-  },
+  "version": "0.0.18",
```
Comment (AlbyIanna) [inline package.json:3]:
```suggestion
  "version": "0.1.0",
```

Comment (AlbyIanna) [reply package.json:3]: We significantly changed the protocol messages, let's bump the minor.

**Comment by AlbyIanna** [comment]
@davegarthsimpson would you please have a look at the comments I made? After that, let's merge this PR right away if no other change is needed.

**Pull request merged**